### ISSUE TITLE
IncompatibleClassChangeError on mvn clean compile

### ISSUE THREAD

**Issue by woj-i** [SimonJPegg/mvn_scalafmt#56]
**Describe the bug**
Incompatibility was encountered while executing org.antipathy:mvn-scalafmt_2.11:1.0.3:format: java.lang.IncompatibleClassChangeError: Method org.scalafmt.interfaces.Scalafmt.create(Ljava/lang/ClassLoader;)Lorg/scalafmt/interfaces/Scalafmt; must be InterfaceMethodref constant

**To Reproduce**
`.scalafmt.config` in the root has:
```version = 2.4.2```

`pom.xml` has: 
``` 
            <plugin>
                <groupId>org.antipathy</groupId>
                <artifactId>mvn-scalafmt_${scala.binary.version}</artifactId>
                <version>1.0.3</version>
                <configuration>
                    <configLocation>${project.basedir}/.scalafmt.conf</configLocation> <!-- path to config -->
                    <skipTestSources>false</skipTestSources> <!-- (Optional) skip formatting test sources -->
                    <skipSources>false</skipSources> <!-- (Optional) skip formatting main sources -->
                    <respectVersion>true</respectVersion> <!-- (Optional) fail if no version set in scalafmt.conf -->
                    <sourceDirectories> <!-- (Optional) Paths to source-directories. Overrides ${project.build.sourceDirectory} -->
                        <param>${project.basedir}/src/main/java</param>
                    </sourceDirectories>
                    <testSourceDirectories> <!-- (Optional) Paths to test-source-directories. Overrides ${project.build.testSourceDirectory} -->
                        <param>${project.basedir}/src/test/java</param>
                    </testSourceDirectories>
                    <validateOnly>true</validateOnly> <!-- check formatting without changing files -->
                    <onlyChangedFiles>true</onlyChangedFiles> <!-- only format (staged) files that have been changed from the specified git branch -->
                    <branch>develop</branch> <!-- The git branch to check against -->
                </configuration>
                <executions>
                    <execution>
                        <phase>validate</phase>
                        <goals>
                            <goal>format</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```

```mvn clean compile```
result:
```/home/wind/repositories/jdk-11.0.2/bin/java -Dmaven.multiModuleProjectDirectory=/home/wind/repositories/ABC -Dmaven.home=/opt/idea-IC-191.7141.44/plugins/maven/lib/maven3 -Dclassworlds.conf=/opt/idea-IC-191.7141.44/plugins/maven/lib/maven3/bin/m2.conf -Dmaven.ext.class.path=/opt/idea-IC-191.7141.44/plugins/maven/lib/maven-event-listener.jar -javaagent:/opt/idea-IC-191.7141.44/lib/idea_rt.jar=45087:/opt/idea-IC-191.7141.44/bin -Dfile.encoding=UTF-8 -classpath /opt/idea-IC-191.7141.44/plugins/maven/lib/maven3/boot/plexus-classworlds-2.6.0.jar org.codehaus.classworlds.Launcher -Didea.version2019.3.3 clean compile
[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------<  >-----------------
[INFO] Building ABC 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ ABC---
[INFO] 
[INFO] --- mvn-scalafmt_2.11:1.0.3:format (default) @ ABC ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.964 s
[INFO] Finished at: 2020-03-11T17:52:21+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.antipathy:mvn-scalafmt_2.11:1.0.3:format (default) on project ABC: Execution default of goal org.antipathy:mvn-scalafmt_2.11:1.0.3:format failed: An API incompatibility was encountered while executing org.antipathy:mvn-scalafmt_2.11:1.0.3:format: java.lang.IncompatibleClassChangeError: Method org.scalafmt.interfaces.Scalafmt.create(Ljava/lang/ClassLoader;)Lorg/scalafmt/interfaces/Scalafmt; must be InterfaceMethodref constant
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.antipathy:mvn-scalafmt_2.11:1.0.3
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/home/wind/.m2/repository/org/antipathy/mvn-scalafmt_2.11/1.0.3/mvn-scalafmt_2.11-1.0.3.jar
[ERROR] urls[1] = file:/home/wind/.m2/repository/org/scala-lang/scala-library/2.11.8/scala-library-2.11.8.jar
[ERROR] urls[2] = file:/home/wind/.m2/repository/javax/enterprise/cdi-api/1.0/cdi-api-1.0.jar
[ERROR] urls[3] = file:/home/wind/.m2/repository/org/eclipse/sisu/org.eclipse.sisu.inject/0.3.3/org.eclipse.sisu.inject-0.3.3.jar
[ERROR] urls[4] = file:/home/wind/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.5.5/plexus-component-annotations-1.5.5.jar
[ERROR] urls[5] = file:/home/wind/.m2/repository/org/codehaus/plexus/plexus-utils/3.1.0/plexus-utils-3.1.0.jar
[ERROR] urls[6] = file:/home/wind/.m2/repository/org/scalameta/scalafmt-dynamic_2.11/2.3.2/scalafmt-dynamic_2.11-2.3.2.jar
[ERROR] urls[7] = file:/home/wind/.m2/repository/org/scalameta/scalafmt-interfaces/2.3.2/scalafmt-interfaces-2.3.2.jar
[ERROR] urls[8] = file:/home/wind/.m2/repository/io/get-coursier/interface/0.0.16/interface-0.0.16.jar
[ERROR] urls[9] = file:/home/wind/.m2/repository/com/typesafe/config/1.4.0/config-1.4.0.jar
[ERROR] urls[10] = file:/home/wind/.m2/repository/org/apache/commons/commons-lang3/3.8/commons-lang3-3.8.jar
[ERROR] urls[11] = file:/home/wind/.m2/repository/commons-io/commons-io/2.6/commons-io-2.6.jar
[ERROR] urls[12] = file:/home/wind/.m2/repository/commons-validator/commons-validator/1.6/commons-validator-1.6.jar
[ERROR] urls[13] = file:/home/wind/.m2/repository/commons-beanutils/commons-beanutils/1.9.2/commons-beanutils-1.9.2.jar
[ERROR] urls[14] = file:/home/wind/.m2/repository/commons-digester/commons-digester/1.8.1/commons-digester-1.8.1.jar
[ERROR] urls[15] = file:/home/wind/.m2/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar
[ERROR] urls[16] = file:/home/wind/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar
[ERROR] urls[17] = file:/home/wind/.m2/repository/junit/junit/3.8.1/junit-3.8.1.jar
[ERROR] urls[18] = file:/home/wind/.m2/repository/org/scala-lang/modules/scala-xml_2.11/1.2.0/scala-xml_2.11-1.2.0.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------```
```
**Expected behavior**
compilation pass

**Environment (please complete the following information):**
    - OS: Linux
    - maven version 3.6.1
    - scalafmt version 2.4.2
    - plugin version 1.0.3
    - openjdk 11.0.2
    - scala 2.11.12

**Comment by bdrillard** [comment]
Also experiencing this issue when using a `maven-compiler-plugin` with compile version greater than Java 1.8.

**Comment by SimonJPegg** [comment]
@woj-i do you see this issue with either of the mvn-scalafmt_2.12 or mvn-scalafmt_2.13 versions?

**Comment by chetanmeh** [comment]
Seeing same issue in our 2.11 project. Just to check would it be ok to use a Scala 2.12 build of plugin to do format check for 2.11 based codebase?

**Comment by SimonJPegg** [comment]
@chetanmeh 2.12 build should be fine.  The plugin has it's own version of the `scala-library` dependency in the jar.  I think this issue might be caused by scala 2.11 on anything over JVM 1.8.  I'm tempted to remove the 2.11 version from future releases.

**Comment by SimonJPegg** [comment]
@woj-i && @chetanmeh does the 2.12 version work for you?

**Comment by woj-i** [comment]
Sorry, I am not able to check that again with 2.12.

**Comment by chetanmeh** [comment]
For us its working fine using 2.12 build with a Scala 2.11 project

**Comment by SimonJPegg** [comment]
Thank @chetanmeh.  I suspect this might be down to the 2.11 version being built with 2.11.8, I've bumped the version to 2.11.12 as recommended [here](https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html), as of [release 2.11-1.0.1594457227.9c01d4f](https://github.com/SimonJPegg/mvn_scalafmt/releases/tag/2.11-1.0.1594457227.9c01d4f).  But, as this can be resolved by using the 2.12 release, I'm closing this issue.
